### ISSUE TITLE
Investigate impact of synchronised module initialization

### ISSUE THREAD

**Issue by SamLex** [Raptoer/F3#197]
I have a suspicion that each module getting initialized concurrently causes more trouble than it is worth. So I would like to investigate the performance and functionality impact of `calling` initialization functions rather than using `execVM`.

I also suspect that `init.sqf` returning before initialization is complete may cause some weird loading issues.

Really, I just want to try it and see what happens because it seems like an odd design choice to me.

**Comment by shadow-fa** [comment]
**Returning before init.sqf is complete**
this will become a problem if we use the suggested `calling` of functions.  (It was unclear if you meant if it currently is a problem or if it will be)

**Component Dependencies**
A lot of components requires f_script_setLocalVars to be done. However, not all of them actually use this line.
Additional dependencies between components:
* orbat requires brefing to be done.
* orbat requires setGroupIDs to be done
* orbat requires assignGear to be done however the waitUntil is missing.
* loadoutNotes requires briefing to be done.
* nametags requires briefing to be done

**Comment by SamLex** [comment]
Sorry, I don't follow. My point is that `init.sqf` *currently* returns ("finishes") before all modules are finished initializing (because they are spawned) and so this might be causing weird issues we haven't really noticed (e.g. getting to the briefing screen before the briefing has been generated, etc.).

Exactly, so instead of `waitUntil` or some other wait method, just initialize (call) briefing before ORBAT and loadoutNotes, etc.

**Comment by shadow-fa** [comment]
`currently returns ("finishes") before all modules are finished`
I remember testing this, and I found it not to be the case.
Once something is spawned (`spawn`, `execVM`), it becomes a completely independent "thread". 
This thread can be waited upon using `scriptDone`, but exiting the caller has no impact on the spawned code.


Doing the briefing before any other component would probably lengthen the execution time of the framework. Even components that do not depend on the briefing in any way could be affected (depending on exact order in `init.sqf`). That being said, I don't think it's noticeable.

However, personally, I'm a fan of explicit marking of dependencies (i.e. `waitUntil X` or `scriptDone Y`.
So I absolutely do not want to lose/remove this. 

Additionally, I suspect upgrading missions would become harder.

In summary, I don't see any advantages in using `call` in init.sqf.
It runs slower and errors in any component could break things for all subsequent `call`s (I'm not 100% sure on this one).
The briefing could be an exception, however even then I would keep explicit checks to make sure things happen (in the right order), even when a missionmaker decides to move things around in init.sqf.

**Comment by shadow-fa** [comment]
As far as I know `init.sqf` starts to run slightly before the briefing shows but of course it also runs during the briefing if it takes longer. However this doesn't cause any issues. If briefing pages are generated later they will still appear whenever they are created. 
(In multiplayer) `init.sqf` runs before and after the Briefing/Map screen, completely independently from that screen.
